elements/rh-pagination/rh-pagination.ts

```diff
@@ -0,0 +1,215 @@
+  </svg>`;
+
+/**
+ * Pagination
+ * @slot - Place element content here
+ */
+@customElement('rh-pagination')
+export class RhPagination extends LitElement {
+  static readonly version = '{{version}}';
+
+  static readonly styles = [styles];
+
+  /**
+   * Override `overflow` values set from HTML or JS.
+   * `overflow` should ideally be private, but because
+   * we can't do `::slotted(nav ol li)`, we need to reflect
+   * it to a host attribute, so that lightdom CSS can target
+   * the list items.
+   */
+  @property({ reflect: true }) overflow: 'start' | 'end' | 'both' | null = null;
+
+  /** Accessible label for the 'nav' element */
+  @property() label = 'Page navigation';
+
+  /** Accessible label for the 'first page' button */
+  @property({ attribute: 'label-first' }) labelFirst = 'first page';
+
+  /** Accessible label for the 'previous page' button */
+  @property({ attribute: 'label-previous' }) labelPrevious = 'previous page';
+
+  /** Accessible label for the 'next page' button */
+  @property({ attribute: 'label-next' }) labelNext = 'next page';
+
+  /** Accessible label for the 'last page' button */
+  @property({ attribute: 'label-last' }) labelLast = 'last page';
+
+  #mo = new MutationObserver(() => this.#update());
+  #screen = new ScreenSizeController(this);
+  #logger = new Logger(this);
+
+  #ol = this.querySelector('ol');
+  #links = this.#ol?.querySelectorAll<HTMLAnchorElement>('li a');
+
+  #currentIndex = -1;
+  #currentLink = this.#getCurrentLink();
+  #firstLink: HTMLAnchorElement | null = null;
+  #lastLink: HTMLAnchorElement | null = null;
+  #nextLink: HTMLAnchorElement | null = null;
+  #prevLink: HTMLAnchorElement | null = null;
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.#mo.observe(this, { childList: true, subtree: true });
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this.#mo.disconnect();
+  }
+
+  update(changed: PropertyValues<this>): void {
+    this.#update();
+    super.update(changed);
+  }
+
+  render() {
+    const { mobile, size } = this.#screen;
+    const { label, labelFirst, labelPrevious, labelNext, labelLast } = this;
+    const firstHref = this.#currentLink === this.#firstLink ? undefined : this.#firstLink?.href;
+    const prevHref = this.#prevLink?.href;
+    const nextHref = this.#nextLink?.href;
+    const lastHref = this.#currentLink === this.#lastLink ? undefined : this.#lastLink?.href;
+    return html`
+      <div id="container" class=${classMap({ mobile, [size as string]: true })}>
+        <a id="first" class="stepper" href=${ifDefined(firstHref)} ?inert=${!firstHref} aria-label=${labelFirst}>${L2}</a>
+        <a id="prev" class="stepper" href=${ifDefined(prevHref)} ?inert=${!prevHref} aria-label=${labelPrevious}>${L1}</a>
```

Comment:
I think we should consider putting part attributes on all of the elements in the shadowroot, even if they are undocumented.  Keep them super flexible if/when the edge-case arises.

Reply:
:grimacing: i don't want to get into a sticky situation here. the design guidelines were pretty clear, and a lot of the functionality is tied to the css. opening up parts could easily break stuff in this case.

Reply:
Gotcha. Sounds like we'll have a fundamental disagreement here. 🙃

I'll mark this as resolved and maybe we can come back to it.